### PR TITLE
Traffic-replay updated

### PR DESCRIPTION
**Pull request by macisasandwich**
- Changed SSL and nonSSL replays to use full paths
- Ignoring connection header since ATS modifies it
- allows custom http methods
- Cleaned up traffic-replay verification logic; prints more useful information if a session fails
- Added in content-length vs actual body length verification #2817 
- SSLReplay and NonSSLReplay now uses the same logic for sending out requests
- Removed unnecessary functions
- Updated h2replay

### CODE REVIEW

File: tests/tools/sessionvalidation/sessionvalidation.py [hunk 1]
```diff
@@ -220,9 +223,9 @@ def validateSingleTransaction(txn):
             # errors by definition can't have bodies
             response_line = txn_resp.getHeaders().split('\r\n')[0]
             response_code = response_line.split(' ')[1]
-            if response_code.startswith('3') and txn_resp.getBody():
-                _verbose_print("transaction response was 3xx and had a body")
-                retval = False
+            # if response_code.startswith('3') and txn_resp.getBody():
+            #     _verbose_print("transaction response was 3xx and had a body")
+            #     retval = False
```
Comment (shinrich) [inline tests/tools/sessionvalidation/sessionvalidation.py]: Why is this commented out?

File: tests/tools/traffic-replay/extractHeader.py [hunk 2]
```diff
@@ -35,14 +35,30 @@ def extract_host(headers):
...
+
+    for key, value in tuples:
+        if key.lower() in ret_dict:
+            ret_dict[key.lower()] = ret_dict[key.lower()] + ", " + value
+        else:
+            ret_dict[key.lower()] = value
+
+    return ret_dict
+
+
```
Comment (shinrich) [inline tests/tools/traffic-replay/extractHeader.py]: Where is this used?

File: tests/tools/sessionvalidation/sessionvalidation.py (hunk 1 above)
Comment (shinrich) [reply tests/tools/sessionvalidation/sessionvalidation.py]: Yes, you should either remove,  add back, or put some comment about why this code is questionable.

**Comment by shinrich** [comment]
Overall the code looks good.  Most of the changes are spacing normalization.  It looks like the main functional changes are adding a execution mode to the microserver (defaulting to test mode).   And fixing the code to verify that the number of body bytes received matches the value of the content length.

**Comment by shinrich** [comment]
[approve ci]

**Comment by macisasandwich** [comment]
@dyrock does this look good to you?

**Comment by dyrock** [comment]
@macisasandwich It has been 3 months but it looks good to me.

**Pull request merged**